lib/active_merchant/billing/gateways/cecabank/cecabank_json.rb

```diff
@@ -51,34 +51,53 @@ def purchase(money, creditcard, options = {})
       end
 
       def void(identification, options = {})
-        authorization, operation_number, money, _network_transaction_id = identification.split('#')
+        authorization, operation_number, money = identification.split('#')
         options[:operation_number] = operation_number
         handle_cancellation(:void, money.to_i, authorization, options)
       end
 
       def refund(money, identification, options = {})
-        authorization, operation_number, _money, _network_transaction_id = identification.split('#')
+        authorization, operation_number, _money = identification.split('#')
         options[:operation_number] = operation_number
         handle_cancellation(:refund, money, authorization, options)
       end
 
       def scrub(transcript)
         before_message = transcript.gsub(%r(\\\")i, "'").scan(/{[^>]*}/).first.gsub("'", '"')
         request_data = JSON.parse(before_message)
-        params =  decode_params(request_data['parametros']).
-                  gsub(%r(("pan\\?"\s*:\s*\\?")[^"]*)i, '\1[FILTERED]').
-                  gsub(%r(("caducidad\\?"\s*:\s*\\?")[^"]*)i, '\1[FILTERED]').
-                  gsub(%r(("cvv2\\?"\s*:\s*\\?")[^"]*)i, '\1[FILTERED]').
-                  gsub(%r(("csc\\?"\s*:\s*\\?")[^"]*)i, '\1[FILTERED]')
-        request_data['parametros'] = encode_params(params)
+        params = parse(request_data['parametros'])
 
+        sensible_fields = decrypt_sensible_fields(params['encryptedData']).
+                          gsub(%r(("pan\\?"\s*:\s*\\?")[^"]*)i, '\1[FILTERED]').
+                          gsub(%r(("caducidad\\?"\s*:\s*\\?")[^"]*)i, '\1[FILTERED]').
+                          gsub(%r(("cvv2\\?"\s*:\s*\\?")[^"]*)i, '\1[FILTERED]').
+                          gsub(%r(("csc\\?"\s*:\s*\\?")[^"]*)i, '\1[FILTERED]').
+                          gsub(%r(("authentication_value\\?"\s*:\s*\\?")[^"]*)i, '\1[FILTERED]')
+
+        params['encryptedData'] = encrypt_sensible_fields(sensible_fields)
+        request_data['parametros'] = encode_params(params.to_json)
         before_message = before_message.gsub(%r(\")i, '\\\"')
         after_message = request_data.to_json.gsub(%r(\")i, '\\\"')
         transcript.sub(before_message, after_message)
       end
 
       private
 
+      def decrypt_sensible_fields(data)
+        cipher = OpenSSL::Cipher.new('AES-256-CBC').decrypt
+        cipher.key = [@options[:sensitive_fields_signature_key]].pack('H*')
```

Comment:
❓ Question:

Wait, you are adding another GSF,  what's the difference between `sensitive_fields_signature_key` and `cypher_key`?

Reply:
Ahhh I can see the  why, 

in this case, the name for the GSF is not entirely correct this is an `encryption key` not a `signature key`,  something short like `encryption_key` should be enough

Reply:
Is the cipher key specific to the merchant or is there one cipher key that will be used for encrypting the required payload on behalf of all Cecabank users?

Reply:
Yes, `cipher.key` is specific for the gateway, it is different to the used for encrypting the payload